doc/api/cli.md

```diff
@@ -253,6 +253,14 @@ effort to report stack traces relative to the original source file.
 Overriding `Error.prepareStackTrace` prevents `--enable-source-maps` from
 modifying the stack trace.
 
+### `--experimental-https-modules`
```

Comment:
i'm hesitant to do this since we don't actually do this for any of the other modules including `--experimental-json-modules`

Reply:
A JSON file is definitely a module, as distinct from a JSON string.

Reply:
> i’m hesitant to do this since we don’t actually do this for any of the other modules

But “https” is not a module. I still think `https-import` is a better name; that’s what you named this PR.

Reply:
I support `https-modules` because https modules cannot access `node:` powerful features so they're different kinds of modules than local modules.

Reply:
I think we could be more clear on this being a consistent decision if we determine we want each dimension (format / url / context / capabilities / etc.) to have its own naming scheme. I'm not a fan of trying to create dimensions inside of flags that are just experimental / not intended to be more than one offs each time so I'd rather we group all the dimensions and have a stable `--experimental-$DIMENSION_VALUE-modules`. It would be complicated to make a new naming scheme for each dimension and would just cause bikeshedding I feel.

Reply:
> I’d rather we group all the dimensions and have a stable `--experimental-$DIMENSION_VALUE-modules`. It would be complicated to make a new naming scheme for each dimension and would just cause bikeshedding I feel.

So for flags ending in `-modules`, the “dimension value” is the module type: `json` or `wasm`, in the cases of `--experimental-json-modules` and `--experimental-wasm-modules`. But `https` is not a module type; it’s the method by which the module is imported. It doesn’t fit with the others.

If you want a similar format for a flag, and you’re anticipating multiple future flags for module import method, you could do `--experimental-https-imports` and later on `--experimental-http-imports` and `--experimental-ftp-imports` and whatever other methods folks think of in the future. Alternatively, you could name the flag `--experimental-network-imports` to cover all protocols, and imports from other sources (like a memory cache maybe) could get their own flags like `--experimental-memory-imports`.

We also aren’t naming experimental flags with any type of namespacing. The existing experimental flags often don’t include the subsystem; see `--experimental-import-meta-resolve` or `--experimental-loader` or `--experimental-specifier-resolution` for modules-related ones that don’t contain “modules” in the name.

Reply:
@GeoffreyBooth there is nothing stating that the flags are actually bound to the dimension type. I actively don't want to spend time battling every dimension flag naming is what I'm saying since they are experimental and subject to change)/ not intended to live forever / that would cause problems for any flag that crosses multiple dimensions like @Jack-Works points out for `https` which covers where the source comes from, the format of the source, and the capabilities of the module. I would want a clear reason why we want to spend time on every feature we add to have this kind of pseudo DSL. Right now I do understand a minor correlation being nice, but it doesn't look like it would be a good idea to me to go down that road since it would have restrictions on what flags could be introduced for example.

Reply:
Per discussion offline, we agreed on `--experimental-network-imports`.